core/templates/dev/head/profile/Profile.js

```diff
@@ -29,22 +31,96 @@ oppia.controller('Profile', [
     $rootScope.loadingMessage = 'Loading';
     $http.get(profileDataUrl).success(function(data) {
       $rootScope.loadingMessage = '';
+      $scope.username = {
+        title: 'Username',
+        value: data.username,
+        helpText: (data.username)
+      };
       $scope.userBio = data.user_bio;
       $scope.userDisplayedStatistics = [{
-        title: 'User Impact Score',
+        title: 'Impact',
         value: data.user_impact_score,
         helpText: (
           'A rough measure of the impact of explorations created by this ' +
           'user. Better ratings and more playthroughs improve this score.')
       }, {
-        title: 'Created Explorations',
+        title: 'Created',
         value: data.created_exp_summary_dicts.length
       }, {
-        title: 'Edited Explorations',
+        title: 'Edited',
         value: data.edited_exp_summary_dicts.length
       }];
-      $scope.userCreatedExplorations = data.created_exp_summary_dicts;
-      $scope.userEditedExplorations = data.edited_exp_summary_dicts;
+
+      $scope.userEditedExplorations = data.edited_exp_summary_dicts.sort(
+          function(exploration1, exploration2) {
+        if (exploration1.ratings > exploration2.ratings) {
+          return 1;
+        } else if (exploration1.ratings === exploration2.ratings) {
+          if (exploration1.playthroughs > exploration2.playthroughs) {
+            return 1;
+          } else if (exploration1.playthroughs > exploration2.playthroughs) {
+            return 0;
+          } else {
+            return -1;
+          }
+        } else {
+          return -1;
+        }
+      });
+
+      $scope.currentPageNumber = 0;
+      $scope.pageSize = 6;
+      $scope.startingExplorationNumber = 1;
+      $scope.endingExplorationNumber = 6;
+      $scope.updateDisplayedExplorationNumbers = function() {
```

Comment:
In the directive template, could you just do the following:

    <[currentPageNumber * pageSize + 1]>

and

    <[Math.min(userEditedExplorations.length, (currentPageNumber + 1) * pageSize)]>

Then you don't need this function.